modules/ingest-common/src/test/java/org/elasticsearch/ingest/common/DotExpanderProcessorTests.java

```diff
@@ -77,16 +76,9 @@ public void testEscapeFields_valueField() throws Exception {
 
         // so because foo is no branch field but a value field the `foo.bar` field can't be expanded
         // into [foo].[bar], so foo should be renamed first into `[foo].[bar]:
+        source.put("foo", new HashMap<>(Map.of("bar", "baz2")));
         IngestDocument document = TestIngestDocument.withDefaultVersion(source);
-        Processor processor = new RenameProcessor(
-            "_tag",
-            null,
-            new TestTemplateService.MockTemplateScript.Factory("foo"),
-            new TestTemplateService.MockTemplateScript.Factory("foo.bar"),
-            false
-        );
-        processor.execute(document);
```

Comment:
I've removed this as the renaming `foo` to `foo.bar` doesn't work anymore since the document already has a `"foo.bar"` key. The rename processor fails if the target already exists. Previously, it was unable to resolve the dotted `"foo.bar" field (`hasField("foo.bar")` returned `false`) but now it can resolve that field and therefore complains that the target field already exists.
I've replaced the usage of the rename processor by simply manually manipulating the source map. As this tests file is about the `DotExpanderProcessor`, not the `RenameProcessor`, that seems fine to me. But it does indicate a subtle change in behavior which may or may not be perceived as breaking.